vmdb/lib/extensions/ar_to_model_hash.rb

```diff
@@ -101,6 +101,12 @@ def to_model_hash_recursive(options, result = nil)
     return result
   end
 
+  def iterable?(klass, association)
+    reflection = klass.virtual_reflection(association) ||
+      klass.reflect_on_association(association)
```

Comment:
@Fryguy I guess I don't want that method.  Since `reflections_with_virtual` merges the hashes, I'd have to directly access the hash (which means I would need to know how the hash is keyed).  Since the keys changed between Rails 3 and 4, I'd like to stick to calling `virtual_reflection` and `reflect_on_association`.

I could add a method to `ar_virtual` like:

```diff
diff --git a/vmdb/lib/extensions/ar_to_model_hash.rb b/vmdb/lib/extensions/ar_to_model_hash.rb
index 9771449..d0f5cd5 100644
--- a/vmdb/lib/extensions/ar_to_model_hash.rb
+++ b/vmdb/lib/extensions/ar_to_model_hash.rb
@@ -102,8 +102,7 @@ module ToModelHash
   end
 
   def iterable?(klass, association)
-    reflection = klass.virtual_reflection(association) ||
-      klass.reflect_on_association(association)
+    reflection = klass.reflection_with_virtual(association)
     reflection && reflection.collection?
   end
 
diff --git a/vmdb/lib/extensions/ar_virtual.rb b/vmdb/lib/extensions/ar_virtual.rb
index cd9de46..5a1eec3 100644
--- a/vmdb/lib/extensions/ar_virtual.rb
+++ b/vmdb/lib/extensions/ar_virtual.rb
@@ -161,6 +161,10 @@ module VirtualFields
     reflections.merge(virtual_reflections)
   end
 
+  def reflection_with_virtual(nams)
+    virtual_reflection(association) || reflect_on_association(association)
+  end
+
 
   #
   # Common methods for Virtual Fields
```

WDYT?